src/composables/transactionItemList.ts

```diff
@@ -0,0 +1,145 @@
+import { computed } from '@vue/composition-api';
+import {
+  IActiveMultisigTx,
+  IDefaultComposableOptions,
+  IDexContracts,
+  ITokenList,
+  ITransaction,
+  ITx,
+} from '../types';
+import {
+  AETERNITY_CONTRACT_ID,
+  defaultTransactionSortingCallback,
+  getInnerTransaction,
+  isTxDex,
+  pipe,
+  TX_DIRECTION,
+} from '../popup/utils';
+import { usePendingMultisigTransaction } from './pendingMultisigTransaction';
+import { useTransactionAndTokenFilter } from './transactionAndTokenFilter';
+import { i18n } from '../store/plugins/languages';
+
+interface TransactionItemList extends IDefaultComposableOptions {
+  isMultisig?: boolean,
+  token: string,
+}
+
+type ITransactionElement = ITransaction | IActiveMultisigTx
+
+export function useTransactionList({
```

Comment:
After spending some time with this code I have a conclusion that it makes no sense to move the code from the component and create separate composable for the code below.
The reason for creating composables is to have a state shared across multiple components. This composable does not have its own state and is included only in one component.

If you want to make the component slimmer you can move most of the filtering functions to utils file.